shared/src/akri/configuration_state.rs

```diff
@@ -0,0 +1,141 @@
+use super::configuration::{Configuration, ConfigurationSpec};
+use log::{error, trace};
+use std::cmp::Ordering;
+
+/// Information for managing the state of a Configuration
+#[derive(Debug, Clone)]
+pub struct ConfigState {
+    /// Tracks the last generation of the `Configuration` resource (i.e. `.metadata.generation`).
+    /// This is used to determine if the `Configuration` actually changed, or if only the metadata changed.
+    /// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+    pub last_generation: Option<i64>,
+    /// The last ConfigurationSpec of this Configuration. Enables tracking changes in events of Configuration modification.
+    pub last_configuration_spec: ConfigurationSpec,
+}
+
+/// In most cases, the Akri Agent handles Configuration changes by deleting and recreating the Configuration.
+/// However, the Agent will not recreate the Configuration if the `.metadata.generation` has NOT changed or if
+/// the only change to the ConfigurationSpec is to the BrokerSpec.
+/// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+/// If only the BrokerSpec changed, the Controller should recreate brokers for the same Configuration.
+pub fn should_recreate_config(config: &Configuration, config_state: &ConfigState) -> bool {
+    match config
+        .metadata
+        .generation
+        .unwrap()
+        .cmp(&config_state.last_generation.unwrap())
+    {
+        Ordering::Less => {
+            error!(
+                "should_recreate_config - configuration generation somehow went backwards {:?} < {:?}",
+                config.metadata.generation, config.metadata.generation
```

Comment:
nit: for this error message, should one of these be config_state.last_generation?